include/seqan3/io/record.hpp

```diff
@@ -246,16 +246,33 @@ struct record : detail::transfer_template_args_onto_t<field_types, std::tuple>
                   "You must give as many IDs as types to seqan3::record.");
 
     //!\brief (Re-)Initialise all tuple elements with `{}`.
-    void clear()
+    void clear() noexcept
     {
         clear_impl(*this, std::make_integer_sequence<size_t, field_types::size()>{});
     }
 private:
-    //!\brief Auxiliary function for clear().
+    //!\brief Auxiliary functions for clear().
     template <size_t ...Is>
-    constexpr void clear_impl(base_type & tup, std::integer_sequence<size_t, Is...> const &)
+    static constexpr void clear_impl(base_type & tup, std::integer_sequence<size_t, Is...> const &) noexcept
```

Comment:
is this really always noexcept? Since there might be aome type that has no clear but does memory allocation when being default constructed

Reply:
Good point. It's unlikely, but not impossible. noexcept should be conditional.